From 32dd02b4cce53db2e39a1ba62806f7db5d152095 Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Mon, 21 Nov 2016 19:10:47 +0100
Subject: debian-sources.list

Forwarded: https://bugs.launchpad.net/cloud-init/+bug/1627293
---
 templates/sources.list.debian.tmpl | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/templates/sources.list.debian.tmpl b/templates/sources.list.debian.tmpl
index c8043f76..f4188c73 100644
--- a/templates/sources.list.debian.tmpl
+++ b/templates/sources.list.debian.tmpl
@@ -26,7 +26,5 @@ deb-src {{mirror}} {{codename}}-updates main contrib non-free
 ## N.B. software from this repository may not have been tested as
 ## extensively as that contained in the main release, although it includes
 ## newer versions of some applications which may provide useful features.
-{#
-deb http://backports.debian.org/debian-backports {{codename}}-backports main contrib non-free
-deb-src http://backports.debian.org/debian-backports {{codename}}-backports main contrib non-free
--#}
+# deb {{mirror}} {{codename}}-backports main contrib non-free
+# deb-src {{mirror}} {{codename}}-backports main contrib non-free
