From c59ea0c01273526baf999cf2d56dfbd44fff3b3b Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Sat, 23 Apr 2016 21:44:34 +0900
Subject: Update syntax of rsyslog snippet ("~" is deprecated).

Bug-Debian: https://bugs.debian.org/791934
Forwarded: https://bugs.launchpad.net/cloud-init/+bug/1367899
---
 tools/21-cloudinit.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/21-cloudinit.conf b/tools/21-cloudinit.conf
index c65325c1..150d800f 100644
--- a/tools/21-cloudinit.conf
+++ b/tools/21-cloudinit.conf
@@ -3,4 +3,4 @@
 
 # comment out the following line to allow CLOUDINIT messages through.
 # Doing so means you'll also get CLOUDINIT messages in /var/log/syslog
-& ~
+& stop
